test/IRGen/opaque_result_type_debug.swift

```diff
@@ -0,0 +1,57 @@
+// RUN: %target-swift-frontend -g -emit-ir -enable-anonymous-context-mangled-names %s | %FileCheck %s
```

Comment:
You should also add tests in test/TypeDecoder, test/RemoteAST and validation-test/Reflection

Reply:
I haven't touched Reflection yet. `swift-remoteast-test` doesn't look very well set up to be able to test this, since there's no way in Swift code to pass a descriptor pointer by itself. Seems like integration-testing this from the lldb side ought to be sufficient?

Reply:
Ah my bad.

I guess swift-remoteast-test will allow you to test the metatype -> mangling -> metatype round trip:

```
printType(type(of: someOpaqueFunc()))
```

Is that useful?

Reply:
Ok, sounds good. However I would still add a TypeDecoder test to test the mangled name to AST type code path in isolation.